app/models/miq_worker.rb

```diff
@@ -317,16 +317,36 @@ def send_message_to_worker_monitor(message, *args)
     )
   end
 
+  def self.preload
+  end
+
+  def self.pre_fork
+    preload
+  end
+
+  def self.after_fork
+    ActiveRecord::Base.connection.reconnect!
```

Comment:
TODO: Remove the `reconnect!` here and in the parent.  Let's see if AR 4.2 does the right thing.